### PR TITLE
Lets fix complete invitation and add create producer to onboarding

### PR DESCRIPTION
**Pull request by andresgutgon**
## TODO
- [x] Fix redirect to signup after logout. Why?
- [x] Fix complete invitation flow
- [x] Fix Re-send invitation
- [x] Implement create producer as onboarding step
## Dependency

https://github.com/coopdevs/katuma/pull/92

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -80,11 +80,11 @@
     "react-helmet": "^2.2.0",
     "react-inline-css": "^2.0.0",
     "react-redux": "4.4.5",
-    "react-router": "2.0.0",
+    "react-router": "2.8.1",
     "react-router-bootstrap": "^0.20.1",
     "react-router-redux": "3.0.0",
     "redux": "^3.0.4",
-    "redux-async-connect": "andresgutgon/redux-async-connect#fix/precompiled-lib",
+    "redux-async-connect": "enricostano/redux-async-connect#fix/precompiled-lib",
```
Comment (enricostano) [inline package.json]: I can open a PR to your fork if you want and keep this as it is.

File: src/components/Button/index.js [hunk 2]
```diff
@@ -92,7 +92,6 @@ const Button = (props) => {
 
 Button.PropTypes = {
   children: PropTypes.any,
-  primary: PropTypes.bool,
```
Comment (enricostano) [inline src/components/Button/index.js]: Check this https://github.com/react-bootstrap/react-bootstrap/issues/2263

**Comment by enricostano** [comment]
@andresgutgon to create a producer you'll need this https://github.com/coopdevs/katuma-web/pull/19

**Comment by enricostano** [comment]
Ehm... I didn't closed this GitHub... I was just removing my commits...

**Comment by enricostano** [comment]
Ehm... I didn't close this GitHub... I was just removing my commits...

**Comment by enricostano** [comment]
Rebased with the new `develop`, so you can start from a balnk slate. Sorry for the noise!

**Comment by enricostano** [comment]
@andresgutgon one thing you'll need in the form that creates producers is the following chain of promises:

``` javascript
  handleSubmit(data) {
    const self = this;

    data.group_id = this.props.group.id;

    // First promise: create the producer
    return this.props.createProducer(data).then((responseData) => {
      const errors = self.props.createProducerErrors;

      if (Object.keys(errors).length) {
        return Promise.reject(errors);
      }

      return Promise.resolve(responseData);
    }).then((producerData) => {
      const supplierData = {
        group_id: data.group_id,
        producer_id: producerData.id,
      };

      // Second promise: create the supplier given the producer ID
      return this.props.createSupplier(supplierData).then(() => {
        const errors = self.props.createSupplierErrors;

        if (Object.keys(errors).length) {
          return Promise.reject(errors);
        }

        return Promise.resolve({});
      }).then(() => {
        // Third promise: load the producers to get latest state
        return this.props.loadProducers(data.group_id).then(() => {

          // TODO: what if the request fails?
          return Promise.resolve({});
        });
      });
    });
  }
```

File: src/containers/Group/Producers/Item.js [hunk 3]
```diff
@@ -1,22 +1,31 @@
...
+    suppliers: PropTypes.object.isRequired,
   }
 
   render() {
-    const { producer } = this.props;
-    const editText = producer.can_edit === true ? 'editar' : '';
-    const providerText = producer.isProvider === true ? 'pedir' : '';
+    const { producer, suppliers } = this.props;
+    const editText = producer.can_edit === true ? 'editar - ' : '';
+    const workingWithThisProducer = isSupplier(producer, suppliers) ? 'si' : 'no';
```
Comment (enricostano) [inline src/containers/Group/Producers/Item.js]: LOL `workingWithThisProducer === provider`

File: src/containers/Group/Producers/Base.js [hunk 4]
```diff
@@ -18,26 +19,64 @@ class GroupProducersBase extends Component {
...
+    super(props);
+
+    this.onOpenModal = this._onOpenModal.bind(this);
+    this.onCloseModal = this._onCloseModal.bind(this);
+
+    this.state = { showModal: false };
+  }
+
+  /**
+   * Open modal with login form
```
Comment (enricostano) [inline src/containers/Group/Producers/Base.js]: login?

File: src/containers/Group/Producers/Base.js [hunk 5]
```diff
@@ -18,26 +19,64 @@ class GroupProducersBase extends Component {
...
+   */
+  _onOpenModal() {
+    this.setState({ showModal: true });
+  }
+
+  /**
+   * Open modal with signup form
```
Comment (enricostano) [inline src/containers/Group/Producers/Base.js]: signup?

File: src/redux/modules/producers/producers.js [hunk 6]
```diff
@@ -131,3 +138,11 @@ export function create(data) {
...
   };
 }
+
+/**
+ * After a new producer is created we reset
+ * in the store `createdProducer` key
+ */
+export function resetCreated() {
+  return { type: RESET_CREATED_PRODUCER };
+}
```
Comment (enricostano) [inline src/redux/modules/producers/producers.js:150]: Why this?

Comment (andresgutgon) [reply src/redux/modules/producers/producers.js:150]: On producer created we set that producer object on `producerCreated` to be used after that for supplier creation (we need producer ID). This action is for cleaning redux state when we finish

File: src/redux/modules/groups/groups.js [hunk 7]
```diff
@@ -50,15 +55,15 @@ export default function groupsReducer(state = initialState, action = {}) {
       };
 
     case LOAD_GROUPS_SUCCESS:
-      entities = action.result;
+      entities = mergeResponse(state.groups.entities, action.result);
```
Comment (enricostano) [inline src/redux/modules/groups/groups.js]: Why not just rewrite the store from scratch every time we fetch the `index` endpoint?

I understand that this maybe be useful while fetching the `show` endpoint, but for `index` I'm worried about data leaks and out of synch data. It seems an early optimization to me.

Comment (andresgutgon) [reply src/redux/modules/groups/groups.js]: Yes, maybe on list we can do that. I'll change it

File: src/redux/modules/producers/producers.js (hunk 6 above)
Comment (enricostano) [reply src/redux/modules/producers/producers.js:150]: why producer reducer have to store state for supplier creation? can we decouple this a bit more?

**Pull request merged**